runtime/compiler/trj9/env/VMJ9.h

```diff
@@ -505,8 +505,7 @@ class TR_J9VMBase : public TR_FrontEnd
    uint32_t                   getAllocationSize(TR::StaticSymbol *classSym, TR_OpaqueClassBlock * clazz);
 
    virtual TR_OpaqueClassBlock *getObjectClass(uintptrj_t objectPointer);
-   virtual uintptrj_t           getReferenceFieldAt(uintptrj_t objectPointer, uintptrj_t offsetFromHeader);
-   virtual uintptrj_t           getVolatileReferenceFieldAt(uintptrj_t objectPointer, uintptrj_t offsetFromHeader);
+   virtual uintptrj_t           getReferenceFieldAt(uintptrj_t objectPointer, uintptrj_t offsetFromHeader, bool isVolatile = false);
    virtual uintptrj_t           getReferenceFieldAtAddress(uintptrj_t fieldAddress);
```

Comment:
In my last comment I said (and still feel), it's cleaner to keep both getReferenceFieldAt and getVolatileReferenceFieldAt separate and thus avoid and extra argument and 'if (isVolatile)' check in getReferenceFieldAt.

But I'm fine with this, too - it's more up to JIT reviewers what prefer more.

Reply:
I would tend to agree with @amicic on this one, especially because `isVolatile` is an optional argument. It may be best to keep the APIs separate so as to show the exact intent of the user. At the very least I'd like to see `isVolatile` not optional so it makes the consumer of the API really think about what they're doing.

Reply:
Thanks for the suggestion, I've reverted the change to getVolatileReferenceFieldAt